modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,752 @@
+      "along with 'duct_block_ids' if 'duct_sizes' is provided.");
+  params.addParam<SubdomainName>(
+      "background_block_name",
+      "Optional customized block name for the background block in 'assembly' mode; must be "
+      "provided along with 'duct_block_names' if 'duct_sizes' is provided.");
+  params.addParam<std::vector<subdomain_id_type>>(
+      "duct_block_ids",
+      "Optional customized block ids for each duct geometry block in 'assembly' mode; must be "
+      "provided along with 'background_block_id'.");
+  params.addParam<std::vector<SubdomainName>>(
+      "duct_block_names",
+      "Optional customized block names for each duct geometry block in 'assembly' mode; must be "
+      "provided along with 'background_block_name'.");
+  params.addRangeCheckedParam<boundary_id_type>("external_boundary_id",
+                                                "external_boundary_id>0",
+                                                "Optional customized external boundary id.");
+  params.addParam<bool>(
+      "create_interface_boundaries", true, "Whether the interface boundary sidesets are created.");
+  params.addParam<std::string>("external_boundary_name",
+                               "Optional customized external boundary name.");
+  params.addParam<bool>("deform_non_circular_region",
+                        true,
+                        "Whether the non-circular region (outside the rings) can be deformed.");
+  params.addParamNamesToGroup("background_block_id background_block_name duct_block_ids "
+                              "duct_block_names external_boundary_id external_boundary_name",
+                              "Customized Subdomain/Boundary");
+  params.addParamNamesToGroup(
+      "generate_control_drum_positions_file assign_control_drum_id position_file", "Control Drum");
+  params.addParamNamesToGroup(
+      "background_intervals duct_intervals uniform_mesh_on_sides deform_non_circular_region",
+      "Mesh Density");
+  params.addClassDescription(
+      "This PatternedCartesianMeshGenerator source code assembles square meshes into a square "
+      "grid "
+      "and optionally forces the outer boundary to be square and/or adds a duct.");
+
+  return params;
+}
+
+PatternedCartesianMeshGenerator::PatternedCartesianMeshGenerator(const InputParameters & parameters)
+  : PolygonMeshGeneratorBase(parameters),
+    _mesh_ptrs(getMeshes("inputs")),
+    _input_names(getParam<std::vector<MeshGeneratorName>>("inputs")),
+    _pattern(getParam<std::vector<std::vector<unsigned int>>>("pattern")),
+    _pattern_boundary(getParam<MooseEnum>("pattern_boundary")),
+    _generate_core_metadata(getParam<bool>("generate_core_metadata")),
+    _background_intervals(getParam<unsigned int>("background_intervals")),
+    _has_assembly_duct(isParamValid("duct_sizes")),
+    _duct_sizes(isParamValid("duct_sizes") ? getParam<std::vector<Real>>("duct_sizes")
+                                           : std::vector<Real>()),
```

Comment:
use a default instead of these conditionals
empty default does convey the information that it s not set

Reply:
The empty default would fail the range check.